internal/cli/cli.go

```diff
@@ -38,83 +42,180 @@ type CommandDefinition struct {
 }
 
 // CommandFactory is a command factory
-type CommandFactory struct {
-	Profile *Profile
-	Config  *Config
+type CommandFactory interface {
+	Build(func() CommandDefinition) *cobra.Command
+	Close()
+	Init()
+	Run(cmd *cobra.Command)
+	SetGlobalFlags(fs *flag.FlagSet)
+}
+
+type commandFactory struct {
+	in      *os.File
+	out     *os.File
+	err     *os.File
```

Comment:
this is odd to me to see `err` as an `*os.File` - can we change this field name at the very least?

Reply:
for sure, I'll add `Reader`/`Writer` as a suffix, since the change in types also stripped that bit of information too